src/ControlSystem/Component.hpp

```diff
@@ -0,0 +1,59 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <tuple>
+
+#include "ControlSystem/Actions/Initialization.hpp"
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Parallel/Actions/SetupDataBox.hpp"
+#include "Parallel/Actions/TerminatePhase.hpp"
+#include "Parallel/Algorithms/AlgorithmSingleton.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/ParallelComponentHelpers.hpp"
+#include "ParallelAlgorithms/Initialization/Actions/RemoveOptionsAndTerminatePhase.hpp"
+#include "Utilities/TMPL.hpp"
+
+/*!
+ * \ingroup ControlSystemGroup
+ * \brief The singleton parallel component responsible for managing a single
+ * control system.
+ *
+ * The control system that this component is responsible for is specified by the
+ * `ControlSystem` template parameter. This control system must conform to the
+ * `control_system::Protocols::ControlSystem` protocol.
+ */
+template <class Metavariables, typename ControlSystem>
+struct ControlComponent {
+  using chare_type = Parallel::Algorithms::Singleton;
+
+  using metavariables = Metavariables;
+
+  using phase_dependent_action_list = tmpl::list<Parallel::PhaseActions<
+      typename metavariables::Phase, metavariables::Phase::Initialization,
+      tmpl::list<Actions::SetupDataBox,
+                 Initialization::Actions::InitializeControlSystem<
+                     Metavariables, ControlSystem>,
+                 Initialization::Actions::RemoveOptionsAndTerminatePhase>>>;
+
+  using initialization_tags = Parallel::get_initialization_tags<
+      Parallel::get_initialization_actions_list<phase_dependent_action_list>>;
+
+  static void execute_next_phase(
+      const typename Metavariables::Phase next_phase,
+      Parallel::CProxy_GlobalCache<Metavariables>& global_cache) noexcept {
+    auto& local_cache = *(global_cache.ckLocalBranch());
+    Parallel::get_parallel_component<
+        ControlComponent<Metavariables, ControlSystem>>(local_cache)
+        .start_phase(next_phase);
+  }
+};
+
+/// \ingroup ControlSystemGroup
+/// List of control componenets to be added to the component list of the
+/// metavars
+template <typename Metavariables, typename ControlSystems>
+using control_components =
+    tmpl::transform<ControlSystems,
+                    tmpl::bind<ControlComponent, Metavariables, tmpl::_1>>;
```

Comment:
`tmpl::pin<Metavariables>`

Reply:
What does `tmpl::pin<>` do?

Reply:
`tmpl::pin` "evaluates to" its argument treated as a constant expression, even if it would normally be treated as a metafunction call.  Types with no template parameters (or *with* non-type template parameters) are always treated as constants, so the effect is only visible if `Metavariables` is a templated struct.  In that case, without the pin Brigand will attempt to call it as a lazy metafunction.